File: plugins/iOS/Badge/Badge.m
//
//  Badge.m
//
//  Author: Smile.fr
//


#import "Badge.h"


@implementation BadgePlugin

@synthesize callbackID;

- (void)set:(NSMutableArray*)arguments withDict:(NSMutableDictionary*)options 
{
	NSString *BadgeNumberString 	= [arguments objectAtIndex:0];
    NSInteger BadgeNumber 	        = [BadgeNumberString integerValue];
	[[UIApplication sharedApplication] setApplicationIconBadgeNumber:BadgeNumber];
    
    //Create Plugin Result
	PluginResult* pluginResult 		= [PluginResult resultWithStatus:PGCommandStatus_OK messageAsString:                        @"Badge set"];
    
    //Call  the Success Javascript function
    [self writeJavascript: [pluginResult toSuccessCallbackString:self.callbackID]];
    
}

- (void)clear:(NSMutableArray*)badgeNumber withDict:(NSMutableDictionary*)options
{
	[[UIApplication sharedApplication] setApplicationIconBadgeNumber:0];
    
    //Create Plugin Result
	PluginResult* pluginResult 		= [PluginResult resultWithStatus:PGCommandStatus_OK messageAsString:                        @"Badge cleared"];
    
    //Call  the Success Javascript function
    [self writeJavascript: [pluginResult toSuccessCallbackString:self.callbackID]];
    
}

@end

File: plugins/iOS/Badge/Badge.js
/**
 * PhoneGap plugin to manage 'Badges' on the app's desktop icon
 *
 * @author Smile.fr
 *
 */
 
BadgePlugin = function() {};

/**
 * sets the badge to 1
 */
BadgePlugin.prototype.set = function(params, success, fail) {
	return PhoneGap.exec( function(args) {
                // handle success
                success(args);
                        },
                        function(args) {
                // handle failure
                fail(args);				 
			}, 'BadgePlugin', 'set', [params]);
};

BadgePlugin.prototype.clear = function(params, success, fail) {
	return PhoneGap.exec( function(args) {
		// handle success
                success(args);
			 }, function(args) {
                // handle failure
                fail(args);
						 
			}, 'BadgePlugin', 'clear', [params]);
};

PhoneGap.addConstructor(function() {
                        if(!window.plugins)
                        {
                                window.plugins = {};
                        }
                                window.plugins.Badge = new BadgePlugin();
});